spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DefaultEnvironmentPostProcessor.java

```diff
@@ -58,7 +64,17 @@ public void postProcessEnvironment(ConfigurableEnvironment environment, SpringAp
 
 		contributeDefaults(internalDefaults, serverDefaultsResource);
 		contributeDefaults(defaults, serverResource);
-
+		String hostName = null;
+		try {
+			hostName = InetAddress.getLocalHost().getHostAddress();
+		}
+		catch (UnknownHostException e) {
+			// ignore
+		}
+		String dataflowServerPort = environment.getProperty("server.port");
+		if (StringUtils.hasText(hostName) && StringUtils.hasText(dataflowServerPort)) {
+			defaults.put(DataFlowPropertyKeys.PREFIX + "server.uri", String.format("http://%s:%s", hostName, dataflowServerPort));
```

Comment:
I thought we don't want to set this value at all and rely only based on whether user has explicitly set this, based on your comment:
>  I think we should check for a spring.cloud.dataflow.server.uri property and if it is available use it, if it's not set just rely on the CTR to use its default.

 but, we can ahead and set it in the environment post processor as well.